### PR TITLE
Slava`s branch

### PR DESCRIPTION
**Pull request by Kotsiuba**

### CODE REVIEW

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+namespace ToDo.ConsoleView
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            CreateWebHostBuilder(args).Build().Run();
+
+            //Item item = new Item();
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: Remove commented code.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using ToDo.DataAccess;
+using Microsoft.EntityFrameworkCore;
+
+namespace ToDo.ConsoleView
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            CreateWebHostBuilder(args).Build().Run();
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: We agreed to create an empty console application. Instead, you've created a web server. Please change it.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+            //Item item = new Item();
+            //Category category = new Category();
+            //Tag tag = new Tag();
+            //TagItem tagItem = new TagItem();
+
+            //FillObjects(item, category, tag, tagItem);
+
+
+            var optionsBuilder = new DbContextOptionsBuilder<ToDoDbContext>();
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: YAGNI

File: ToDo.DataAccess/ToDo.ConsoleView/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using ToDo.DataAccess.DataBase;
+using Microsoft.EntityFrameworkCore;
+using ToDo.DataAccess;
+
+namespace ToDo.ConsoleView
+{
+    public class Startup
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/Startup.cs]: Remove this file at all.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 5]
```diff
@@ -5,23 +5,24 @@
...
 {
-    public class GenericRepository<T> : IDataRepository<T> where T : class
+    public class GenericRepository : IDataRepository
     {
         private readonly ToDoDbContext _context;
 
         public GenericRepository(ToDoDbContext context)
         {
             _context = context;
+            ToDoDbInitializer.Initialize(_context);
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Remove initialization from here.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 6]
```diff
@@ -5,23 +5,24 @@
 
 namespace ToDo.DataAccess
 {
-    public class GenericRepository<T> : IDataRepository<T> where T : class
+    public class GenericRepository : IDataRepository
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: No need to change repository.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 7]
```diff
@@ -5,23 +5,24 @@
...
         private readonly ToDoDbContext _context;
 
         public GenericRepository(ToDoDbContext context)
         {
             _context = context;
+            ToDoDbInitializer.Initialize(_context);
         }
 
-        public void Add(T entity)
+        public void Add<T>(T entity) where T : class
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: No need to add constraints here.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]
This is absolutely another from what I wanted. We agreed about an empty console application. There mustn't be changes in layer implemented before. Please change it. 

JFY. Please name branches from noun of what you do in it.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs (hunk 1 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: Removed.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs (hunk 2 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: I created, but didn't use it. Deleted.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs (hunk 3 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: Changed.

File: ToDo.DataAccess/ToDo.ConsoleView/Startup.cs (hunk 4 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/Startup.cs]: Removed.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 7 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Removed.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 6 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Changed back.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 5 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Removed.

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 8]
```diff
@@ -10,7 +10,7 @@ public class GenericRepository<T> : IDataRepository<T> where T : class
         private readonly ToDoDbContext _context;
 
         public GenericRepository(ToDoDbContext context)
-        {
+        {
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Fix

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 9]
```diff
@@ -36,9 +36,11 @@ public void Edit(T entity)
             throw new NotImplementedException();
         }
 
+
+
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Fix

File: ToDo.DataAccess/ToDo.DataAccess/IDataRepository.cs [hunk 10]
```diff
@@ -8,5 +8,6 @@ public interface IDataRepository<T>
         IEnumerable<T> GetAll();
         bool Delete(T entity);
         void Edit(T entity);
+
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/IDataRepository.cs]: Fix

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs [hunk 11]
```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.AspNetCore.Hosting;
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: Remove redundant usings

File: ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj [hunk 12]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.6" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.3" />
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj]: Remove redundant packages

File: ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.6" />
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj]: Remove redundant packages

**Review by zhikin2207 — CHANGES_REQUESTED** [review]
Almost

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 8 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Fixed

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 9 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Fixed

File: ToDo.DataAccess/ToDo.DataAccess/IDataRepository.cs (hunk 10 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/IDataRepository.cs]: fixed

File: ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj (hunk 12 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj]: Unistalled

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs [hunk 14]
```diff
@@ -35,10 +35,10 @@ public void Edit(T entity)
         {
             throw new NotImplementedException();
         }
-
+
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Remove spaces.

File: ToDo.DataAccess/ToDo.ConsoleView/Program.cs (hunk 11 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/Program.cs]: Removed.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]

File: ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj (hunk 13 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.ConsoleView/ToDo.ConsoleView.csproj]: Removed

File: ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs (hunk 14 above)
Comment (Kotsiuba) [reply ToDo.DataAccess/ToDo.DataAccess/GenericRepository.cs]: Removed.

File: ToDo.DataAccess/ToDo.DataAccess/ToDo.DataAccess.csproj [hunk 15]
```diff
@@ -115,6 +115,7 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="IDataRepository.cs" />
```
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDo.DataAccess.csproj]: You don't have this file right now.

**Review by zhikin2207 — CHANGES_REQUESTED** [review]

File: ToDo.DataAccess/ToDo.DataAccess/ToDo.DataAccess.csproj (hunk 15 above)
Comment (zhikin2207) [inline ToDo.DataAccess/ToDo.DataAccess/ToDo.DataAccess.csproj]: You don't have this fucking file!

**Review by zhikin2207 — CHANGES_REQUESTED** [review]

**Review by zhikin2207 — APPROVED** [review]

**Pull request merged**